kmod/src/data.c

```diff
@@ -1804,6 +1993,7 @@ const struct address_space_operations scoutfs_file_aops = {
 	.writepages		= scoutfs_writepages,
 	.write_begin		= scoutfs_write_begin,
 	.write_end		= scoutfs_write_end,
+	.direct_IO		= scoutfs_direct_IO,
```

Comment:
There's a scary subtle land mine here.  We use the .aio_write and .aio_read methods because they're the only reaonsable generic file io wrappers available.   Calling blockdev_direct_IO from aio_write can actually use aio and return with IOs in flight, which we don't support.   (We do post IO processing as .direct_IO returns, not in dio completion.. that'd be a whole other thing.).   So as we add direct io, we want to be sure that we're disabling AIO.   The only way I can see to do that is to add something like if (!is_sync_kiocb(iocb)) return -EINVAL to the top of our scoutfs_file_aio_{read,write}.